ChildNode/ParentNode no longer extending Node

According to https://github.com/microsoft/TypeScript/issues/45266, there are some breaks around [`ChildNode` no longer being a subtype of `Node`](https://github.com/microsoft/TypeScript/commit/9ebe11c2d375bfb1c8fad10d705ffae09583c522#diff-dc0eab937d15e62545da3ed7b4f40ad6b24f15dd88fbc6ceda2bfb4ed8356eb0L3511). This was also mentioned in Google's 4.4 beta adoption thread (#1067).

<details>
<summary>
Some breaks here
</summary>

> # [ant-design/ant-design](https://github.com/ant-design/ant-design)
> ### [tsconfig.json](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/tsconfig.json)
>  - `error TS2359: The right-hand side of an 'instanceof' expression must be of type 'any' or of a type assignable to the 'Function' interface type.`
>    - [components/_util/scrollTo.ts#L26](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/_util/scrollTo.ts#L26)
>  - `error TS2339: Property 'nodeType' does not exist on type 'ChildNode'.`
>    - [components/typography/util.tsx#L143](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L143)
>    - [components/typography/util.tsx#L207](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L207)
>  - `error TS2345: Argument of type 'ChildNode' is not assignable to parameter of type 'Node'.`
>    - [components/typography/util.tsx#L160](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L160)
>    - [components/typography/util.tsx#L165](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L165)
>    - [components/typography/util.tsx#L220](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L220)
>  - `error TS2339: Property 'textContent' does not exist on type 'ChildNode'.`
>    - [components/typography/util.tsx#L227](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L227)

> # [sindresorhus/refined-github](https://github.com/sindresorhus/refined-github)
> ### [tsconfig.json](https://github.com/sindresorhus/refined-github/blob/4af1669e2dba94a1d170dfe563b88416582be716/tsconfig.json)
>  - `error TS2345: Argument of type 'Element | ChildNode' is not assignable to parameter of type 'string | Node'.`
>    - [source/helpers/dom-utils.ts#L48](https://github.com/sindresorhus/refined-github/blob/4af1669e2dba94a1d170dfe563b88416582be716/source/helpers/dom-utils.ts#L48)

</details>

In #1067, @saschanaz explained

> It's a mixin interface implemented by some Node interfaces. The correct way to receive Nodes that can be children is `ChildNode & Node`, just like `ParentNode & Node` for Nodes that can be parents.

I may be missing some context of why that's important to have - was there discussion around that? But regardless, I don't know if we can retroactively make that call. If that's a thing users need, does it make sense to create a `ChildNodeBase` and a `ParentNodeBase`?

--- a/baselines/dom.generated.d.ts
+++ b/baselines/dom.generated.d.ts
@@ -10471,11 +10471,11 @@ interface Node extends EventTarget {
     /**
      * Returns the children.
      */
-    readonly childNodes: NodeList;
+    readonly childNodes: NodeListOf<ChildNode>;
     /**
      * Returns the first child.
      */
-    readonly firstChild: Node | null;
+    readonly firstChild: ChildNode | null;
     /**
      * Returns true if node is connected and false otherwise.
      */
@@ -10483,11 +10483,11 @@ interface Node extends EventTarget {
     /**
      * Returns the last child.
      */
-    readonly lastChild: Node | null;
+    readonly lastChild: ChildNode | null;
     /**
      * Returns the next sibling.
      */
-    readonly nextSibling: Node | null;
+    readonly nextSibling: ChildNode | null;
     /**
      * Returns a string appropriate for the type of node.
      */
@@ -10508,11 +10508,11 @@ interface Node extends EventTarget {
     /**
      * Returns the parent.
      */
-    readonly parentNode: Node | null;
+    readonly parentNode: ParentNode | null;
     /**
      * Returns the previous sibling.
      */
-    readonly previousSibling: Node | null;
+    readonly previousSibling: ChildNode | null;
     textContent: string | null;
     appendChild<T extends Node>(node: T): T;
     /**

--- a/baselines/dom.generated.d.ts
+++ b/baselines/dom.generated.d.ts
@@ -3463,7 +3463,7 @@ declare var CharacterData: {
     new(): CharacterData;
 };
 
-interface ChildNode {
+interface ChildNode extends Node {
     /**
      * Inserts nodes just after node, while replacing strings in nodes with equivalent Text nodes.
      *
@@ -10471,11 +10471,11 @@ interface Node extends EventTarget {
     /**
      * Returns the children.
      */
-    readonly childNodes: NodeListOf<ChildNode & Node>;
+    readonly childNodes: NodeList;
     /**
      * Returns the first child.
      */
-    readonly firstChild: (ChildNode & Node) | null;
+    readonly firstChild: Node | null;
     /**
      * Returns true if node is connected and false otherwise.
      */
@@ -10483,11 +10483,11 @@ interface Node extends EventTarget {
     /**
      * Returns the last child.
      */
-    readonly lastChild: (ChildNode & Node) | null;
+    readonly lastChild: Node | null;
     /**
      * Returns the next sibling.
      */
-    readonly nextSibling: (ChildNode & Node) | null;
+    readonly nextSibling: Node | null;
     /**
      * Returns a string appropriate for the type of node.
      */
@@ -10508,11 +10508,11 @@ interface Node extends EventTarget {
     /**
      * Returns the parent.
      */
-    readonly parentNode: (Node & ParentNode) | null;
+    readonly parentNode: Node | null;
     /**
      * Returns the previous sibling.
      */
-    readonly previousSibling: (ChildNode & Node) | null;
+    readonly previousSibling: Node | null;
     textContent: string | null;
     appendChild<T extends Node>(node: T): T;
     /**
@@ -10929,7 +10929,7 @@ declare var PannerNode: {
     new(context: BaseAudioContext, options?: PannerOptions): PannerNode;
 };
 
-interface ParentNode {
+interface ParentNode extends Node {
     readonly childElementCount: number;
     /**
      * Returns the child elements.

--- a/inputfiles/overridingTypes.jsonc
+++ b/inputfiles/overridingTypes.jsonc
@@ -2,11 +2,9 @@
     "mixins": {
         "mixin": {
             "ChildNode": {
-                "name": "ChildNode",
                 "extends": "Node"
             },
             "ParentNode": {
-                "name": "ParentNode",
                 "extends": "Node"
             },
             "Animatable": {

--- a/inputfiles/overridingTypes.jsonc
+++ b/inputfiles/overridingTypes.jsonc
@@ -1,6 +1,14 @@
 {
     "mixins": {
         "mixin": {
+            "ChildNode": {
+                "name": "ChildNode",
+                "extends": "Node"
+            },
+            "ParentNode": {
+                "name": "ParentNode",
+                "extends": "Node"
+            },
             "Animatable": {
                 "name": "Animatable",
                 "methods": {
@@ -760,26 +768,8 @@
                 },
                 "properties": {
                     "property": {
-                        "parentNode": {
-                            "overrideType": "Node & ParentNode"
-                        },
                         "parentElement": {
                             "overrideType": "HTMLElement"
-                        },
-                        "childNodes": {
-                            "overrideType": "NodeListOf<ChildNode & Node>"
-                        },
-                        "firstChild": {
-                            "overrideType": "ChildNode & Node"
-                        },
-                        "lastChild": {
-                            "overrideType": "ChildNode & Node"
-                        },
-                        "nextSibling": {
-                            "overrideType": "ChildNode & Node"
-                        },
-                        "previousSibling": {
-                            "overrideType": "ChildNode & Node"
                         }
                     }
                 }

--- a/inputfiles/overridingTypes.jsonc
+++ b/inputfiles/overridingTypes.jsonc
@@ -768,8 +768,26 @@
                 },
                 "properties": {
                     "property": {
+                        "parentNode": {
+                            "overrideType": "ParentNode"
+                        },
                         "parentElement": {
                             "overrideType": "HTMLElement"
+                        },
+                        "childNodes": {
+                            "overrideType": "NodeListOf<ChildNode>"
+                        },
+                        "firstChild": {
+                            "overrideType": "ChildNode"
+                        },
+                        "lastChild": {
+                            "overrideType": "ChildNode"
+                        },
+                        "nextSibling": {
+                            "overrideType": "ChildNode"
+                        },
+                        "previousSibling": {
+                            "overrideType": "ChildNode"
                         }
                     }
                 }

